tests/test_ssh_context.py

```diff
@@ -107,6 +107,14 @@ def test_upload(self):
 
     def test_download(self):
         self.context.download(self.__class__.submission)
+
+    def test_tar_compress(self):
+        self.machine.context.remote_profile['tar_compress'] = False
```

Comment:
This is not correct for the following reason:
- `self.machine` is defined in the `setUpClass` and this only executes once for the entire class. When you change the object from `setUpClass`,  other test methods will be affected.
- `remote_profile` is only used to initialize the `SSHSession`. See:
https://github.com/deepmodeling/dpdispatcher/blob/3ec2e4a4d933093b3eedd4092f8f5d2069a84d15/dpdispatcher/ssh_context.py#L300